ansible_collections/arista/avd/plugins/action/eos_validate_state_runner.py

```diff
@@ -0,0 +1,111 @@
+
+        save_catalog = self._task.args.get("save_catalog")
+        if not isinstance(save_catalog, bool):
+            raise AnsibleActionFail(f"'save_catalog' must be a boolean, got {save_catalog}.")
+
+        save_catalog_name = None
+        if save_catalog:
+            if device_catalog_output_dir := self._task.args.get("device_catalog_output_dir"):
+                save_catalog_name = f"{device_catalog_output_dir}/{hostname}-catalog.yml"
+            else:
+                raise AnsibleActionFail("'device_catalog_output_dir' must be set when 'save_catalog' is True.")
+
+        skipped_tests = self._task.args.get("skipped_tests", [])
+        # TODO once we have pydantic, check the list elements
+        if not isinstance(skipped_tests, list):
+            raise AnsibleActionFail(f"'skipped_tests' must be a list of dictionnaries, got {skipped_tests}.")
+
+        # Fetching ansible tags for backward compatibility
+        ansible_tags = {
+            "ansible_run_tags": task_vars.get("ansible_run_tags", ()),
+            "ansible_skip_tags": task_vars.get("ansible_skip_tags", ()),
+        }
+
+        # This is not all the hostvars, but just the Ansible Hostvars Manager object where we can retrieve hostvars for each host on-demand.
+        hostvars = task_vars["hostvars"]
+
+        try:
+            anta_device = AnsibleEOSDevice(name=hostname, connection=ansible_connection, check_mode=ansible_check_mode)
+            anta_results = get_anta_results(
+                anta_device=anta_device,
+                hostvars=hostvars,
+                logging_level=logging_level,
+                skipped_tests=skipped_tests,
+                ansible_tags=ansible_tags,
+                save_catalog_name=save_catalog_name,
+                # This convert Ansible Check Mode to dry_run
+                dry_run=ansible_check_mode,
+                yaml_dumper=AnsibleNoAliasDumper,
+            )
+        except Exception as error:
+            raise AnsibleActionFail(message=str(error)) from error
+
+        result["results"] = anta_results
+
+        return result
+
+
+def setup_module_logging(hostname: str, result: dict) -> None:
+    """
+    Create a Filter to add the hostname to generate logs,
+    Add a Handler to copy the logs from the plugin into Ansible output based on their level
+
+    Args:
+      hostname (str): Current Inventory device being used to augment the logs with <hostname>
+      result (dict): The ansible dictionary used for the results
+    """
+    python_to_ansible_filter = PythonToAnsibleContextFilter(hostname)
+    python_to_ansible_handler = PythonToAnsibleHandler(result, display)
+    python_to_ansible_handler.addFilter(python_to_ansible_filter)
+    LOGGER.addHandler(python_to_ansible_handler)
+    # TODO mechanism to manipulate the logger globally for pyavd
+    # Keep debug to be able to see logs with `-v` and `-vvv`
+    LOGGER.setLevel(logging.DEBUG)
```

Comment:
This needs to be addressed in future improvement - maybe taking the verbosity level of ansible to select the level of logging.